src/rand.rs

```diff
@@ -432,3 +435,34 @@ mod fuchsia {
         fn zx_cprng_draw(buffer: *mut u8, length: usize);
     }
 }
+
+#[cfg(any(target_env = "sgx"))]
+mod sgx {
+    use core::arch::x86_64::_rdrand64_step;
+    use crate::error;
+
+    #[inline]
+    fn rdrand() -> Result<u64, error::Unspecified> {
+        // See https://software.intel.com/en-us/articles/intel-digital-random-number-generator-drng-software-implementation-guide
+        // Section 5.2.1
+        for _ in 0..10 {
+            let mut r = 0;
+            if unsafe { _rdrand64_step(&mut r) } == 1 {
+                return Ok(r);
+            }
+        }
+
+        Err(error::Unspecified)
+    }
+
+    pub fn fill(mut dest: &mut [u8]) -> Result<(), error::Unspecified> {
+        while dest.len() > 0 {
```

Comment:
This can me made more clear (and more efficient by eliding `memcpy` calls) by using [`chunks_exact_mut`](https://doc.rust-lang.org/core/slice/struct.ChunksExactMut.html), like the `getrandom` implementation does.  

See the comparison here: https://rust.godbolt.org/z/gwprxd